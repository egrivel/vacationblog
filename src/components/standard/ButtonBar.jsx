'use strict';

const React = require('react');

const Formrow = require('./Formrow.jsx');

const ButtonBar = React.createClass({
  displayName: 'ButtonBar',

  propTypes: {
    buttons: React.PropTypes.array.isRequired
  },

  render: function() {
    const buttonList = [];
    for (let i = 0; i < this.props.buttons.length; i++) {
      if (i) {
        buttonList.push(' ');
      }
      buttonList.push(
        <button
          key={'b-' + i}
          onClick={this.props.buttons[i].onClick}
        >
          {this.props.buttons[i].label}
        </button>
      );
    }
    return (
      <Formrow
        label=""
      >
        <div>
          {buttonList}
        </div>
      </Formrow>
    );
  }
});

module.exports = ButtonBar;
