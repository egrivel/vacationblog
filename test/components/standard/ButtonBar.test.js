import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import ButtonBar from '../../../src/components/standard/ButtonBar';

/*
 * Test strategy
 * The component gets an array of one or more buttons. If there are
 * multiple buttons, they are separated by a space.
 */
describe('components/standard/ButtonBar', () => {
  test('renders single button', () => {
    const buttonList = [
      {
        label: 'Test Label',
        onClick: sinon.stub()
      }
    ];
    const render = renderer.create(
      <ButtonBar buttons={buttonList} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('renders multiple buttons', () => {
    const buttonList = [
      {
        label: 'Test Label 1',
        onClick: sinon.stub()
      },
      {
        label: 'Test Label 2',
        onClick: sinon.stub()
      }
    ];
    const render = renderer.create(
      <ButtonBar buttons={buttonList} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
