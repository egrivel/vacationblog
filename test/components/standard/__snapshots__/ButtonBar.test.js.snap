// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/standard/ButtonBar #render renders multiple buttons 1`] = `
<div>
  <div
    className="formLabel"
  >
    
  </div>
  <div
    className="formValue"
  >
    <div>
      <button
        onClick={[Function]}
        type="submit"
      >
        Test Label 1
      </button>
       
      <button
        onClick={[Function]}
        type="submit"
      >
        Test Label 2
      </button>
    </div>
  </div>
</div>
`;

exports[`components/standard/ButtonBar #render renders single button 1`] = `
<div>
  <div
    className="formLabel"
  >
    
  </div>
  <div
    className="formValue"
  >
    <div>
      <button
        onClick={[Function]}
        type="submit"
      >
        Test Label
      </button>
    </div>
  </div>
</div>
`;
